crates/parser/src/jup_programs/orca/ix_parser/parser.rs

```diff
@@ -0,0 +1,140 @@
+                    },
+                ))
+            },
+            SWAP_V2_IX_DISC => {
+                check_min_accounts_req(accounts_len, 15)?;
+                let swap_ix_v2_data: SwapV2IxData =
+                    BorshDeserialize::deserialize(&mut ix_data).unwrap();
+                Ok(OrcaProgramIx::SwapV2(
+                    SwapV2Accounts {
+                        token_program_a: ix.accounts[0],
+                        token_program_b: ix.accounts[1],
+                        memo_program: ix.accounts[2],
+                        token_authority: ix.accounts[3],
+                        whirlpool: ix.accounts[4],
+                        token_mint_a: ix.accounts[5],
+                        token_mint_b: ix.accounts[6],
+                        token_owner_account_a: ix.accounts[7],
+                        token_vault_a: ix.accounts[8],
+                        token_owner_account_b: ix.accounts[9],
+                        token_vault_b: ix.accounts[10],
+                        tick_array0: ix.accounts[11],
+                        tick_array1: ix.accounts[12],
+                        tick_array2: ix.accounts[13],
+                        oracle: ix.accounts[14],
+                    },
+                    SwapV2IxData {
+                        a_to_b: swap_ix_v2_data.a_to_b,
+                        amount: swap_ix_v2_data.amount,
+                        other_amount_threshold: swap_ix_v2_data.other_amount_threshold,
+                        sqrt_price_limit: swap_ix_v2_data.sqrt_price_limit,
+                        amount_specified_is_input: swap_ix_v2_data.amount_specified_is_input,
+                    },
+                ))
+            },
+            _ => Err(ParseError::from("Unknown instruction")),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use yellowstone_vixen_mock::{run_ix_parse, tx_fixture, FixtureData, LoadFixtureFilters};
+
+    use super::*;
+
+    #[tokio::test]
+    async fn test_swap_ix_parsing() {
+        let parser = OrcaProgramIxParser;
+
+        let filters = LoadFixtureFilters {
+            outer_ixs_programs: vec![
+                "JUP6LkbZbjS1jKKwapdHNy74zcZ3tLUZoi5QNyVTaV4".to_string(),
+                orca_whirlpools_client::ID.to_string(),
+            ],
+            inner_ixs_discriminators: vec![SWAP_IX_DISC],
+        };
+
+        let ixs = tx_fixture!("3WC8LGHHs3wYzWef1YmLsRS96G1s5BV4XJYhzvypgWp1uGG16SxepFCd7FhHaTieW66Yn9JFR4tUPA1HYArgFZaA",
+            Some(filters));
+
+        let parsed = run_ix_parse!(parser, &ixs[0].inner[0]);
+        if let OrcaProgramIx::Swap(accounts, data) = parsed {
+            assert_eq!(
+                accounts.whirlpool.to_string(),
+                "ENYEHSyduTbFN1xoSEGaLu7c1F8AqKucdscMuV5Yypy2".to_string()
+            );
+            assert_eq!(data.a_to_b, false);
+        } else {
+            panic!("Invalid Instruction");
+        }
+    }
```

Comment:
```suggestion
        let ix = instruction_fixture!("3WC8LGHHs3wYzWef1YmLsRS96G1s5BV4XJYhzvypgWp1uGG16SxepFCd7FhHaTieW66Yn9JFR4tUPA1HYArgFZaA", 0, 1);

        let parsed = run_ix_parse!(parser, &ix);
        if let OrcaProgramIx::Swap(accounts, data) = parsed {
            assert_eq!(
                accounts.whirlpool.to_string(),
                "ENYEHSyduTbFN1xoSEGaLu7c1F8AqKucdscMuV5Yypy2".to_string()
            );
            assert_eq!(data.a_to_b, false);
        } else {
            panic!("Invalid Instruction");
        }
    }
```

What do you think of this dx? Caller provides the signature and path to the instruction they want. This example is the second inner instruction of the second instruction of the transaction.

We then write just that instruction to the disk at path like `ix-{sig}-0-1`

Reply:
Can be done , but these indexes need to be know before itself , so users need to search for these indexes beforehand